### PR TITLE
SW-4755 Airbyte DoorDash connector

### PR DESCRIPTION
**Pull request by wesley-ks**
Tried customizing a low code connector first but that didn't work out to well. It actually makes it very difficult to follow what happens when. 

Pivoted to use the Python API template. Most of what I learned is documented here:

https://philzcoffee.atlassian.net/wiki/spaces/SW/pages/2646933505/Creating+a+new+custom+connector

Majority of the code comes from the generator and there's no need to spend time reviewing them. The only file of interest is source.py. Even then it's not a large file -- the time was mainly spent on figuring out what property/method to override. It wasn't too helpful to look through other connectors, as the implementations are vastly different, and it's hard to tell which one has the best practices.

Luckily, the DoorDash API is rather straightforward, saved for the need to wait for the report to be ready and to extract the CSV from a zip. Their documentation is not 100% accurate, but after working with Uber Eats, NetSuite, and Paylocity, this is nothing to complain about. 😄

### CODE REVIEW

**Review by tayeke — APPROVED** [review]
Wow that does not look like anything was easy there. I'm most curious why the data you're getting is zipped?

**Comment by wesley-ks** [comment]
Thanks Taylor. Yeah, no idea why they would do it that way instead of a compressed stream.

File: airbyte-integrations/connectors/source-doordash/source_doordash/source.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        self.logger.info(f'Report ready for download. URL: {report_link}.')
+        report_download_response = requests.get(report_link, headers=self.authenticator.get_auth_header())
+        with tempfile.TemporaryDirectory() as temp_dir:
+            with tempfile.TemporaryFile(dir=temp_dir, suffix='.zip') as report_zip_file:
+                report_zip_file.write(report_download_response.content)
+                report_zip_file.flush()
+                self.logger.info(f'Report download. Zip file written to {report_zip_file.name}.')
+
+                # Unzip the zip file.
+                with zipfile.ZipFile(report_zip_file, 'r') as zip_obj:
```
Comment (todesignandconquer) [inline airbyte-integrations/connectors/source-doordash/source_doordash/source.py:117]: What are the performance implications of a lot of this unarchiving?

File: airbyte-integrations/connectors/source-doordash/source_doordash/source.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                    file_list = zip_obj.namelist()
+                    for file_name in file_list:
+                        csv_dest = os.path.join(temp_dir, file_name)
+                        self.logger.info(f'Extracting file {csv_dest}.')
+                        zip_obj.extract(file_name, path=temp_dir)
+
+                        # Load the extracted CSV.
+                        with open(csv_dest, 'r', encoding='utf_8') as f:
```
Comment (todesignandconquer) [inline airbyte-integrations/connectors/source-doordash/source_doordash/source.py:125]: Likewise, how many rows are being read through? If it's a lot there could be more efficient ways to pull this data out.

**Review by todesignandconquer — APPROVED** [review]
Some questions around performance...

File: airbyte-integrations/connectors/source-doordash/source_doordash/source.py (hunk 1 above)
Comment (wesley-ks) [reply airbyte-integrations/connectors/source-doordash/source_doordash/source.py:117]: Yeah good points. The whole thing is based on how DoorDash's API works, so we don't really have any other options. Fortunately, the export is rather small (60MB). They hold a max of 2 years of data. With us starting in 2022, I think we will probably max out at 200MB.

It comes across as a single zip file that would unzip into a single CSV.

I don't know if there's any other efficient way to unzip and read the CSV in Python, so please let me know if you have any suggestions.

Comment (todesignandconquer) [reply airbyte-integrations/connectors/source-doordash/source_doordash/source.py:117]: @tayeke Would pandas be a potential performance improvement here?

**Pull request merged**